api/provider/aws/test_aws/helpers.go

```diff
@@ -52,3 +55,55 @@ func deleteSGHelper(mockAWS *awsc.MockClient, securityGroupID string) {
 		DeleteSecurityGroup(input).
 		Return(&ec2.DeleteSecurityGroupOutput{}, nil)
 }
+
+func healthCheckHelper(healthCheck *models.HealthCheck) *elb.HealthCheck {
+	if healthCheck == nil {
+		elbHealthCheck := &elb.HealthCheck{}
+		elbHealthCheck.SetTarget("TCP:80")
+		elbHealthCheck.SetInterval(int64(30))
+		elbHealthCheck.SetTimeout(int64(5))
+		elbHealthCheck.SetHealthyThreshold(int64(2))
+		elbHealthCheck.SetUnhealthyThreshold(int64(2))
+
+		return elbHealthCheck
+	}
+
+	elbHealthCheck := &elb.HealthCheck{}
+	elbHealthCheck.SetTarget(healthCheck.Target)
+	elbHealthCheck.SetInterval(int64(healthCheck.Interval))
+	elbHealthCheck.SetTimeout(int64(healthCheck.Timeout))
+	elbHealthCheck.SetHealthyThreshold(int64(healthCheck.HealthyThreshold))
+	elbHealthCheck.SetUnhealthyThreshold(int64(healthCheck.UnhealthyThreshold))
+
+	return elbHealthCheck
+}
+
+func listenerHelper(port *models.Port) *elb.Listener {
+	if port == nil {
+		listener := &elb.Listener{}
+		listener.SetProtocol("tcp")
+		listener.SetLoadBalancerPort(80)
+		listener.SetInstancePort(80)
+		listener.SetInstanceProtocol("tcp")
+
+		return listener
+	}
+
+	listener := &elb.Listener{}
+	listener.SetProtocol(port.Protocol)
+	listener.SetLoadBalancerPort(port.HostPort)
+	listener.SetInstancePort(port.ContainerPort)
+
+	switch strings.ToLower(port.Protocol) {
+	case "http", "https":
+		listener.SetInstanceProtocol("http")
+	case "tcp", "ssl":
+		listener.SetInstanceProtocol("tcp")
+	}
+
+	if port.CertificateName != "" {
```

Comment:
Doesn't the certificate name need to be resolved first? I'm guessing this is passing only because you specify the certificate name == arn below. That's too much magic, we need to be more explicit about that mapping and ensure the caller uses the arn/name in the correct places.